core/core-rest/RestClient.cpp

```diff
@@ -641,6 +643,32 @@ void RestClient::authenticateRefresh(
     NLOG_ERROR("exception: " + string(e.what()));
   }
 }
+void RestClient::sessionLogout(
+    NSessionPtr session,
+    std::function<void()> successCallback,
+    ErrorCallback errorCallback) {
+  try {
+    RestReqContext* ctx = createReqContext(nullptr);
+    setSessionAuth(ctx, session);
+
+    ctx->successCallback = successCallback;
+    ctx->errorCallback = errorCallback;
+
+    NHttpQueryArgs args;
+
+    rapidjson::Document document;
+    document.SetObject();
+
+    document.AddMember("token", session->getAuthToken(), document.GetAllocator());
```

Comment:
`setSessionAuth` above sets `Authorisation` HTTP header. Why do we need token in the payload?

Reply:
Oh, interesting. Looking at the code there could be multiple valid tokens and it can invalidate just those passed in request body. If nothing passed it logs out user from all active sessions.